extensions/amp-3d-gltf/0.1/test/integration/test-amp-3d-gltf.js

```diff
@@ -0,0 +1,42 @@
+/**
+ * Copyright 2018 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {poll} from '../../../../../testing/iframe';
+
+describe('amp-3d-gltf', function() {
+  describes.integration('amp-3d-gltf iframe renders', {
+    body: '<amp-3d-gltf layout="fixed"' +
+                      ' width="200"' +
+                      ' height="200"' +
+                      ' src="/123"></amp-3d-gltf>',
+    extensions: ['amp-3d-gltf'],
+  }, () => {
+
+    it('iframe renders', () => {
+      const loadPromise = waitForAnimationLoad();
+      return loadPromise.then(iframeExists => {
+        expect(iframeExists).to.be.true;
+      });
+    });
+  });
```

Comment:
Could you add a test verifying that the animation is properly paused when the element is out of view?

Reply:
Currently there is no need to check if animation is running from main component. Animation only exists inside 3p-iframe.
If I am going to do this test I'll have to add this logic. Is there any way to write code that will only work in test environment?

Reply:
Maybe a test intercepts the postMessage and ensures `pause` is sent?

Reply:
Yes that is what I did eventually.